From acf3bf94d03e5e27c51859c1ea91c943ed5de2be Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Fri, 18 Jan 2013 10:51:07 +0100
Subject: [PATCH 56/70] CLEANUP: config: set the maxaccept value for peers
 listeners earlier

Since we introduced bind_conf in peers, we can set maxaccept in a cleaner
way at the proper time, let's do this to make the code more readable.
---
 src/cfgparse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cfgparse.c b/src/cfgparse.c
index 1cdda34..6b1034e 100644
--- a/src/cfgparse.c
+++ b/src/cfgparse.c
@@ -1757,6 +1757,7 @@ int cfg_parse_peers(const char *file, int linenum, char **args, int kwm)
 				}
 
 				list_for_each_entry(l, &bind_conf->listeners, by_bind) {
+					l->maxaccept = 1;
 					l->maxconn = ((struct proxy *)curpeers->peers_fe)->maxconn;
 					l->backlog = ((struct proxy *)curpeers->peers_fe)->backlog;
 					l->timeout = &((struct proxy *)curpeers->peers_fe)->timeout.client;
@@ -7076,7 +7077,6 @@ out_uri_auth_compat:
 		while (*last) {
 			curpeers = *last;
 			if (curpeers->peers_fe) {
-				LIST_NEXT(&curpeers->peers_fe->conf.listeners, struct listener *, by_fe)->maxaccept = 1;
 				last = &curpeers->next;
 				continue;
 			}
-- 
1.8.3.2

